packages/xstate-test/src/TestModel.ts

```diff
@@ -85,15 +85,21 @@ export class TestModel<TState, TEvent extends EventObject> {
     return this.getPaths({ ...options, pathGenerator: getShortestPaths });
   }
 
+  private _getStatePaths(
+    options?: Partial<GetPathsOptions<TState, TEvent>>
+  ): Array<StatePath<TState, TEvent>> {
+    const pathGenerator =
+      options?.pathGenerator || TestModel.defaults.pathGenerator;
+    return pathGenerator(this.behavior, this.resolveOptions(options));
+  }
+
   public getPaths(
     options?: Partial<GetPathsOptions<TState, TEvent>>
   ): Array<TestPath<TState, TEvent>> {
-    const pathGenerator = pathGeneratorWithDedup<TState, TEvent>(
-      options?.pathGenerator || TestModel.defaults.pathGenerator
-    );
-    const paths = pathGenerator(this.behavior, this.resolveOptions(options));
-
-    return paths.map(this.toTestPath);
+    return deduplicatePaths(
+      this._getStatePaths(options),
+      options?.serializeEvent
+    ).map(this.toTestPath);
   }
 
   public getShortestPathsTo(
```

Comment:
q: I somewhat don't get why `getShortestPathsTo` does its own filtering here while `getSimplePathsTo` is able to do leverage filtering implemented in `traverseSimplePathsTo`. The distinction seems to be a little bit odd.